compiler/rustc_borrowck/src/type_check/free_region_relations.rs

```diff
@@ -233,16 +234,29 @@ impl<'tcx> UniversalRegionRelationsBuilder<'_, 'tcx> {
 
             // In the new solver, normalize the type-outlives obligation assumptions.
             if self.infcx.next_trait_solver() {
-                match deeply_normalize(
-                    self.infcx.at(&ObligationCause::misc(span, defining_ty_def_id), param_env),
-                    outlives,
-                ) {
-                    Ok(normalized_outlives) => {
-                        outlives = normalized_outlives;
-                    }
-                    Err(e) => {
-                        self.infcx.err_ctxt().report_fulfillment_errors(e);
-                    }
+                let Ok(TypeOpOutput {
+                    output: normalized_outlives,
+                    constraints: constraints_normalize,
+                    error_info: _,
+                }) = CustomTypeOp::new(
+                    |ocx| {
+                        ocx.deeply_normalize(
+                            &ObligationCause::dummy_with_span(span),
+                            self.param_env,
+                            outlives,
+                        )
+                        .map_err(|_: Vec<ScrubbedTraitError<'tcx>>| NoSolution)
+                    },
+                    "normalize type outlives obligation",
+                )
+                .fully_perform(self.infcx, span)
+                else {
+                    self.infcx.dcx().delayed_bug(format!("could not normalize {outlives:?}"));
+                    continue;
+                };
+                outlives = normalized_outlives;
+                if let Some(c) = constraints_normalize {
+                    constraints.push(c);
                 }
```

Comment:
can you move this into a sub-fn

The function is already quite complex imo and this code isn't straightforward (I don't think it can be made less so, really)